File: edgequake/sdks/swift/Sources/EdgeQuakeSDK/EdgeQuakeClient.swift
import Foundation

/// Main client for the EdgeQuake API.
///
/// Usage:
/// ```swift
/// let client = EdgeQuakeClient(config: EdgeQuakeConfig(baseUrl: "http://localhost:8080"))
/// let health = try await client.health.check()
/// print(health.status ?? "unknown")
/// ```
public final class EdgeQuakeClient: @unchecked Sendable {
    private let http: HttpHelper

    public let health: HealthService
    public let documents: DocumentService
    public let entities: EntityService
    public let relationships: RelationshipService
    public let graph: GraphService
    public let query: QueryService
    public let chat: ChatService
    public let tenants: TenantService
    public let users: UserService
    public let apiKeys: ApiKeyService
    public let tasks: TaskService
    public let pipeline: PipelineService
    public let models: ModelService
    public let costs: CostService
    public let conversations: ConversationService
    public let folders: FolderService
    public let lineage: LineageService
    // OODA-35: New services
    public let auth: AuthService
    public let workspaces: WorkspaceService
    public let shared: SharedService

    public init(config: EdgeQuakeConfig = EdgeQuakeConfig()) {
        self.http = HttpHelper(config: config)
        self.health = HealthService(http)
        self.documents = DocumentService(http)
        self.entities = EntityService(http)
        self.relationships = RelationshipService(http)
        self.graph = GraphService(http)
        self.query = QueryService(http)
        self.chat = ChatService(http)
        self.tenants = TenantService(http)
        self.users = UserService(http)
        self.apiKeys = ApiKeyService(http)
        self.tasks = TaskService(http)
        self.pipeline = PipelineService(http)
        self.models = ModelService(http)
        self.costs = CostService(http)
        self.conversations = ConversationService(http)
        self.folders = FolderService(http)
        self.lineage = LineageService(http)
        // OODA-35: New services
        self.auth = AuthService(http)
        self.workspaces = WorkspaceService(http)
        self.shared = SharedService(http)
    }
}

File: edgequake/sdks/csharp/src/EdgeQuakeSDK/EdgeQuakeClient.cs
namespace EdgeQuakeSDK;

/// <summary>Main client for the EdgeQuake API.</summary>
/// <remarks>OODA-36: Added AuthService, WorkspaceService, SharedService.</remarks>
public class EdgeQuakeClient
{
    public HealthService Health { get; }
    public DocumentService Documents { get; }
    public EntityService Entities { get; }
    public RelationshipService Relationships { get; }
    public GraphService Graph { get; }
    public QueryService Query { get; }
    public ChatService Chat { get; }
    public TenantService Tenants { get; }
    public UserService Users { get; }
    public ApiKeyService ApiKeys { get; }
    public TaskService Tasks { get; }
    public PipelineService Pipeline { get; }
    public ModelService Models { get; }
    public CostService Costs { get; }
    public ConversationService Conversations { get; }
    public FolderService Folders { get; }
    /// <summary>WHY: Lineage service added in OODA-24 for entity/document/chunk provenance.</summary>
    public LineageService Lineage { get; }
    /// <summary>WHY: Auth service added in OODA-36 for login/logout/refresh/me.</summary>
    public AuthService Auth { get; }
    /// <summary>WHY: Workspace service added in OODA-36 for workspace CRUD/stats/switch.</summary>
    public WorkspaceService Workspaces { get; }
    /// <summary>WHY: Shared service added in OODA-36 for conversation sharing.</summary>
    public SharedService Shared { get; }

    public EdgeQuakeClient(EdgeQuakeConfig? config = null)
    {
        config ??= new EdgeQuakeConfig();
        var http = new HttpHelper(config);

        Health = new HealthService(http);
        Documents = new DocumentService(http);
        Entities = new EntityService(http);
        Relationships = new RelationshipService(http);
        Graph = new GraphService(http);
        Query = new QueryService(http);
        Chat = new ChatService(http);
        Tenants = new TenantService(http);
        Users = new UserService(http);
        ApiKeys = new ApiKeyService(http);
        Tasks = new TaskService(http);
        Pipeline = new PipelineService(http);
        Models = new ModelService(http);
        Costs = new CostService(http);
        Conversations = new ConversationService(http);
        Folders = new FolderService(http);
        Lineage = new LineageService(http);
        Auth = new AuthService(http);
        Workspaces = new WorkspaceService(http);
        Shared = new SharedService(http);
    }
}
